tests/interfaces/default_qubit_2_integration/test_tensorflow_autograph_qnode_shot_vector_default_qubit_2.py

```diff
@@ -0,0 +1,469 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Integration tests for using the TF interface with shot vectors and with a QNode"""
+# pylint: disable=too-many-arguments,too-few-public-methods,unexpected-keyword-arg
+import pytest
+
+import pennylane as qml
+from pennylane import numpy as np
+from pennylane import qnode
+from pennylane.devices.experimental import DefaultQubit2
+
+pytestmark = pytest.mark.tf
+
+tf = pytest.importorskip("tensorflow")
+
+shots_and_num_copies = [((1, (5, 2), 10), 4)]
+shots_and_num_copies_hess = [((10, (5, 1)), 2)]
+
+qubit_device_and_diff_method = [
+    [DefaultQubit2(), "finite-diff", {"h": 10e-2}],
+    [DefaultQubit2(), "parameter-shift", {}],
+    [DefaultQubit2(), "spsa", {"h": 10e-2, "num_directions": 20}],
+]
+
+TOLS = {
+    "finite-diff": 0.3,
+    "parameter-shift": 1e-2,
+    "spsa": 0.3,
+}
+
+
+@pytest.mark.parametrize("shots,num_copies", shots_and_num_copies)
+@pytest.mark.parametrize("dev,diff_method,gradient_kwargs", qubit_device_and_diff_method)
+@pytest.mark.parametrize(
+    "decorator,interface",
+    [(tf.function, "tf"), (lambda x: x, "tf-autograph")],
+)
+class TestReturnWithShotVectors:
+    """Class to test the shape of the Grad/Jacobian/Hessian with different return types and shot vectors."""
+
+    def test_jac_single_measurement_param(
+        self, dev, diff_method, gradient_kwargs, shots, num_copies, decorator, interface
+    ):
+        """For one measurement and one param, the gradient is a float."""
+
+        @decorator
+        @qnode(dev, diff_method=diff_method, interface=interface, **gradient_kwargs)
+        def circuit(a, **_):
```

Comment:
from the PR description^: I also give **_ to functions jitted with tf.function if they need shots, otherwise it fails because of an unexpected kwarg (shots).

Reply:
Thanks!